### PR TITLE
Using schema registry with Aiven for Apache Kafka

### PR DESCRIPTION
**Pull request by snuyanzin**
# What changed, and why it matters

This is a draft version for #475 

I also added a ready to use java/maven based example for that article. The problem is that it has it's own file structure and I don't know how to put it into `code/products`. For that reason now it is in a separate repo https://github.com/snuyanzin/schema-registry-with-avro-example
In case there is a better place for that, I can move it there

// cc @ftisiot

### CODE REVIEW

**Comment by ftisiot** [comment]
Thanks @snuyanzin for your contribution! I have a couple of quick feedbacks before checking the details of the article:
* Can your code example be incorporated in https://github.com/aiven/aiven-examples?
* Can we reduce the amount of images? Instead of visually showing the Overview page with parameters, please describe the components needed, good examples are [1](https://developer.aiven.io/docs/products/kafka/howto/kafka-tools-config-file.html) and [2](https://developer.aiven.io/docs/products/kafka/howto/get-topic-partition-details.html). Reducing the number of screenshots will help us to reduce the work in the future in case our UI changes.

Thanks again for the help!

**Comment by snuyanzin** [comment]
> Can your code example be incorporated in https://github.com/aiven/aiven-examples?

yes, I made a PR https://github.com/aiven/aiven-examples/pull/35

**Comment by ftisiot** [comment]
Hi @snuyanzin, I changed the wording a bit to be more descriptive, can you verify if it's still correct? 
I would also wait for the PR for the aiven-examples to get merged before merging this doc, WDYT?

**Comment by snuyanzin** [comment]
Hi @ftisiot , thanks for corrections, it looks great.
Agree about waiting for PR.
I would also say that after the PR will be merged it makes sense to update this PR as well since currently it refers to code example in PR, not in main/master branch

**Comment by ftisiot** [comment]
Yep exactly! Do you know who can we ask for a review on the aiven-example side?

**Comment by snuyanzin** [comment]
I've just asked Ryan (in PR) from OSPO team, he is a committer and PMC in Avro, so probably he is the best candidate to be a reviewer .
Also I could  ping other OSPO team members

File: docs/products/kafka/howto/schema-registry.rst
Comment (reta) [inline docs/products/kafka/howto/schema-registry.rst:80]: I think would be useful to include expected output over there (fe `*.class` files)

File: docs/products/kafka/howto/schema-registry.rst
Comment (reta) [inline docs/products/kafka/howto/schema-registry.rst:107]: And here as well, just to highlight what is going to appear in `${project.basedir}/src/main/generated-sources/` once the plugin does the generation step.

File: docs/products/kafka/howto/schema-registry.rst
Comment (snuyanzin) [reply docs/products/kafka/howto/schema-registry.rst:80]: the issue is that in fact it generates lots of code for that (about 1k lines....)
I will try to see if it is possible to add here small but significant code-snippet from the whole result of generation

Comment (reta) [reply docs/products/kafka/howto/schema-registry.rst:80]: Oh, I mean something like `When the generation finishes successfully, the `ClickRecord.java` should appear <somewhere>`, just to make expectations explicit.

Comment (snuyanzin) [reply docs/products/kafka/howto/schema-registry.rst:80]: > a Java class ``ClickRecord.java`` in the ``io.aiven.avro.example`` package
I guess it is mentioned here.
Or probably makes sense to mentioned that the package structure also will be generated in case of absence

Comment (snuyanzin) [reply docs/products/kafka/howto/schema-registry.rst:80]: done

File: docs/products/kafka/howto/schema-registry.rst
Comment (snuyanzin) [reply docs/products/kafka/howto/schema-registry.rst:109]: done

**Review by reta — APPROVED** [review]

**Comment by snuyanzin** [comment]
@ftisiot fyi:
https://github.com/aiven/aiven-examples/pull/35 is merged and i updated link to master

**Comment by ftisiot** [comment]
Thanks @snuyanzin! Added minor modification, ready to be reviewed by other devrel members

File: docs/products/kafka/howto/schema-registry.rst
Comment (RyanSkraba) [inline docs/products/kafka/howto/schema-registry.rst:28]: This is a bit out of sync with the code sample -- they aren't placeholders, they're command-line arguments (which is far better, in my opinion!)

Comment (snuyanzin) [reply docs/products/kafka/howto/schema-registry.rst:28]: this could be considered as a placeholder when a reader copy-pastes it directly from here.
At the same time the example in aiven-examples is ready to run example... 
However agree, probably need to add more clarification here

Comment (ftisiot) [reply docs/products/kafka/howto/schema-registry.rst:28]: What are the next steps here? are we ready to merge?
The "placeholder" is a standard term across our devportal used across a lot of pages [1](https://developer.aiven.io/docs/products/redis/howto/connect-node.html), [2](https://developer.aiven.io/docs/products/postgresql/howto/create-manual-backups.html) as examples

Comment (snuyanzin) [reply docs/products/kafka/howto/schema-registry.rst:28]: I added a comment that it's needed when a user wants to implement producer/consumer on his/her own
i think it should be enough

Comment (RyanSkraba) [reply docs/products/kafka/howto/schema-registry.rst:28]: OK, I see -- in that case, I don't think we need to worry too much about clarification, the intention is pretty obvious with the code sample at aiven-examples and the snippets here.  Thanks!

File: docs/products/kafka/howto/schema-registry.rst
Comment (RyanSkraba) [inline docs/products/kafka/howto/schema-registry.rst:119]: My apologies, I think it was clearer without this line!  Since the code snippets and the aiven-example project are presented here at the same time and side-by-side, it's unnecessary.

File: docs/products/kafka/howto/schema-registry.rst
Comment (snuyanzin) [reply docs/products/kafka/howto/schema-registry.rst:28]: ok, thanks, then resolve this

**Review by TibsAtWork — APPROVED** [review]
Text seems good, thank you.

**Pull request merged**
